### ISSUE TITLE
refactor CommitInstantiated to a "method" of registry entry

### ISSUE THREAD

**Issue by dherman** [whatwg/loader#79]
CommitInstantiated is a weird and confusing abstract operation. Its real purpose is to flesh out the internal state of a registry entry record. This would be a little easier to understand if it were a spec "method" of a registry entry record. Also, it should have a better name.

To be clear, registry entry records are *not* JS objects, so I'm not talking about a JS method. But ECMA 262 does have a precedent for internal records of the semantics having methods on them.

Something like:
```
1. Let _status_ be _entry_.ExtractInstanceDependencies(_loader_, _optionalInstance_, _source_).
```
or maybe split it into two separate operations like:
```
1. Let _status_ be _entry_.ExtractInstance(_loader_, _optionalInstance_, _source_).
1. ReturnIfAbrupt(_status_).
1. _entry_.ExtractDependencies().
```
Something like that...

**Comment by caridy** [comment]
Could they be just abstract methods of Registry Objects? Just like we did for Module Objects?

**Comment by caridy** [comment]
This is part of https://github.com/whatwg/loader/pull/97

**Comment by caridy** [comment]
fixed by #97
